my_list.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'enumerable'
+
+class MyList
+  include MyEnumerable
+
+  def initialize(*items)
+    @list = items
+  end
+
+  def each(&block)
+    i = 0
+    while i < @list.size
+      block.call(@list[i])
+      i += 1
+    end
+  end
```

Comment:
- [ ] Please implement the `#each` method that `yields` all members of `@list` in succession but in your case, you didn't use the `yield` method in your code Kindly use the `yield` method in your code because Yield is a keyword (meaning it's a core part of the language) & it's used inside methods for calling a block. In other words, yield is pausing our method & transferring control to the block so that it can do its thing & then come back with a return value.:slightly_smiling_face: 

**Example:**
```suggestion
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```